syft/workers/base.py

```diff
@@ -911,62 +911,95 @@ def _fetch_protocol_remote(
 
         return None
 
-    def search(self, query: Union[List[Union[str, int]], str, int]) -> List[PointerTensor]:
+    def search(self, query: Union[List[Union[str, int]], str, int]) -> List:
         """Search for a match between the query terms and a tensor's Id, Tag, or Description.
 
         Note that the query is an AND query meaning that every item in the list of strings (query*)
         must be found somewhere on the tensor in order for it to be included in the results.
 
         Args:
             query: A list of strings to match against.
-            me: A reference to the worker calling the search.
 
         Returns:
-            A list of PointerTensors.
+            A list of valid results found.
+
+        TODO Search on description is not supported for the moment
         """
         if isinstance(query, (str, int)):
             query = [query]
+        # Empty query returns all the tagged and registered values
+        elif len(query) == 0:
+            results = []
+            for obj in self._objects.values():
+                if hasattr(obj, "tags") and obj.tags is not None and len(obj.tags) > 0:
+                    results.append(obj)
+            return results
+
+        results = None
+        for query_item in query:
+            results_item = set()
```

Comment:
Why do you create this intermediary set instead of putting everything into results?

Reply:
good point